### DISCUSSION TITLE
Using import.meta.glob in a module definition ends in error : "SyntaxError: Cannot use 'import.meta' outside a module"

### DISCUSSION THREAD

**Discussion by BrunoETH** [nuxt/nuxt#27808] [Questions]
Hello !

I'm traying to build a module inside a Nuxt app where I'd like to use the Vite feature 'import.meta.glob' to load some JSON files like this "/path/to/**/*.json" but when i start the app I get a "SyntaxError : Cannot use 'import.meta' outside a module".

I'd like to know if it is allowed to use this Vite feature in the Nuxt modules ? If so, what is the configuration to get this works ? (note that, it's works fine everywhere else).

nuxt.config.ts
```
export default defineNuxtConfig({
  devtools: { enabled: true },
  build: {
    transpile: ["@nuxt/typescript-build"],
  },
  vite: {
    esbuild: {
      target: "esnext",
    },
  },
  modules: ["/modules/myModule"],
});
```

tsconfig.json
```
{
  // https://nuxt.com/docs/guide/concepts/typescript
  "extends": "./.nuxt/tsconfig.json",
  "compilerOptions": {
    "moduleResolution": "node",
    "types": ["node", "@nuxt/types", "vite/client"],
    "baseUrl": "."
  }
}
```

modules/my-module.ts
```
import { defineNuxtModule } from "@nuxt/kit";

export default defineNuxtModule({
  meta: {
    name: "my-Module",
    configKey: "myModule",
  },
  setup(_, nuxt) {
    const files = import.meta.glob("/public/data2/**/*.json", { eager: true });
    console.log("Module loaded successfully");
  },
});
```

Here is a [reproduction link](https://codesandbox.io/p/devbox/compassionate-hertz-vfqvp6?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxqqapk500053j6k2z1gy07b%2522%252C%2522sizes%2522%253A%255B58.17330210772833%252C41.82669789227167%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxqqapk400023j6k4d4w8cw4%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxqqapk400033j6krb59vbnk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxqqapk500043j6k3t6ogcut%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxqqapk400023j6k4d4w8cw4%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxqqapk400013j6kqnff6lu4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxqqewfj001t3j6kb620u8rc%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Ftsconfig.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxqqapk400023j6k4d4w8cw4%2522%252C%2522activeTabId%2522%253A%2522clxqqewfj001t3j6kb620u8rc%2522%257D%252C%2522clxqqapk500043j6k3t6ogcut%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clxqqapk500043j6k3t6ogcut%2522%257D%252C%2522clxqqapk400033j6krb59vbnk%2522%253A%257B%2522id%2522%253A%2522clxqqapk400033j6krb59vbnk%2522%252C%2522activeTabId%2522%253A%2522clxtmnpe8003p3j6jgqpfy2ip%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxqrbot1010n3j6kl25jist6%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clxqrbov70433dig71py5hl3d%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clxtmnpe8003p3j6jgqpfy2ip%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) 

Thank you.

**Comment by ndragun92** [comment]
In that case you can use file system and do it in such way

Example: (Feel free to modify it based on your needs)
```
const fs = require("fs");
const path = require("path");
import { defineNuxtModule } from "@nuxt/kit";

const directory = path.join(__dirname, "../public/data2");

function readJsonFilesRecursively(dir) {
  fs.readdir(dir, { withFileTypes: true }, (err, files) => {
    if (err) {
      console.error("Error reading directory:", err);
      return;
    }

    files.forEach((file) => {
      const filePath = path.join(dir, file.name);

      if (file.isDirectory()) {
        // If the file is a directory, recurse into it
        readJsonFilesRecursively(filePath);
      } else if (file.isFile() && path.extname(file.name) === ".json") {
        // If the file is a JSON file, read and parse it
        fs.readFile(filePath, "utf8", (err, data) => {
          if (err) {
            console.error("Error reading file:", filePath, err);
            return;
          }

          try {
            const json = JSON.parse(data);
            console.log("Content of", filePath, ":", json);
          } catch (parseErr) {
            console.error("Error parsing JSON in file:", filePath, parseErr);
          }
        });
      }
    });
  });
}

export default defineNuxtModule({
  meta: {
    name: "my-Module",
    configKey: "myModule",
  },
  setup(_, nuxt) {
    //const files = import.meta.glob("/public/data2/**/*.json", { eager: true });
    const test = readJsonFilesRecursively(directory);
    console.log("Module loaded successfully", test);
  },
});

```

**Reply by BrunoETH** [reply]
Hello !

Thanks for your replay.
So, do you think it's a bug or a kind of not to do things ?

As a workaround I installed the "[glob](https://github.com/isaacs/node-glob#readme)" package and it's working perfectly.

```
import { defineNuxtModule } from "@nuxt/kit";
import { glob } from "glob";

export default defineNuxtModule({
  meta: {
    name: "my-Module",
    configKey: "myModule",
  },
  setup(_, nuxt) {
    const files = await glob("/public/data2/**/*.json");
    console.log("Module loaded successfully");
  }
});
```

**Reply by ndragun92** [reply]
Yea I think that glob wasnt available in such context. Installing packages for that could actually solve issue with being able to use it
